experimental/LieAlgebras/src/LieAlgebraModule.jl

```diff
@@ -46,6 +46,11 @@ base_ring(V::LieAlgebraModule{C}) where {C<:RingElement} = base_ring(base_lie_al
 
 base_ring(v::LieAlgebraModuleElem{C}) where {C<:RingElement} = base_ring(parent(v))
```

Comment:
Not about this PR but about existing code: I find this surprising, I would have expected `base_ring` to return the Lie algebra. But instead it seems to return the base ring of the Lie algebra, i.e. its "coefficient ring" or "left-acting domain" (in GAP parlance). Of course in part this is because `base_ring` is notoriously ill-defined, so guessing what it does is always problematic.

And right now, this definition is inconsistent with other existing module types, where the full acting ring is returned.

So this is a bit unfortunate. One solution then would be to change `base_ring(V::LieAlgebraModule)` to return the same as `base_lie_algebra(V)`, and then add a separate function to access the "coefficients domain". 

E.g. under the name `coefficient_ring`; while it is not super common to refer to the `R` in `R-algebra` by the name `coefficient ring`, I think it is fine as long as we specify it explicitly: 
> The coefficient ring of a module is the coefficient ring of its base ring.
> The coefficient ring of a ring which is *constructed* as an R-algebra is R.

The bit about "constructed as" is meant to clarify that while e.g. the polynomial ring `GF(4)[:x]` is both an $F_4$ and an $F_2$ algebra, it was constructed as an $F_4$ algebra.

After typing this I remembered we have #1505 (and also #938) ... so there :-)

Reply:
I understand your point.

However, I found it quite confusing to refer to the lie algebra as `base_ring`, although it is not a ring. My idea now would be to abolish `base_ring`, as it is misleading and instead use `base_lie_algebra` and `coefficient_ring`.
But that will happen in a future PR.

Reply:
Well, every Lie algebra is also a Lie ring, and many people refer to "non-associative rings". But yeah, a more generic name than `base_ring` would be helpful...

But in general my view remains that I'd wish we could do completely away with `base_ring` ;-). And if we have `coefficient_ring` and `base_lie_algebra` (or, for that matter, just `lie_algebra(M)` ???), it's not so important what `base_ring` does exactly (or we could even consider omitting it entirely?).

Reply:
I think for the name of genericity (is this a word), the name should not contain "lie". For the modules over finite-dimensional associative algebras, I called this just `algebra(M)` and `coefficient_ring(M) == coefficient_ring(algebra(M))` (there is no `base_ring(M)`).
```
julia> A = matrix_algebra(QQ, 2);

julia> V = Hecke.Amodule(A, [matrix(a) for a in basis(A)]);

julia> algebra(V)
Matrix algebra of dimension 4 over Rational field

julia> coefficient_ring(V)
Rational field
```
Not sure about the `coefficient_ring` anymore, but I like `algebra(M)`. I think it would if we could try to be somehow consistent. (Sorry about the missing documentation.)

Reply:
With `algebra`, I have kind of the same problem as with `base_ring` for denoting the Lie algebra, as it is not an algebra in the usual sense.

Reply:
We just relax what we mean with an "algebra", so it could be fine. I don't think this would be a big problem. But we will discuss this later.